homeassistant/components/shelly/climate.py

```diff
@@ -186,3 +292,41 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
             await self.set_state_full_path(
                 schedule=1, schedule_profile=f"{preset_index}"
             )
+
+    async def async_added_to_hass(self) -> None:
+        """Handle entity which will be added."""
+        _LOGGER.info("Restoring entity %s", self.name)
+
+        last_state = await self.async_get_last_state()
+
+        if last_state is not None:
+            self.last_state = last_state
+            self.last_state_attributes = self.last_state.attributes
+            self._preset_modes = cast(
+                list, self.last_state.attributes.get("preset_modes")
+            )
+
+        self.async_on_remove(self.wrapper.async_add_listener(self._update_callback))
+
+    async def async_update(self) -> None:
+        """Update entity with latest info."""
+        await self.wrapper.async_request_refresh()
```

Comment:
Why don't we inherit from CoordinatorEntity so we don't need to implement the same methods and registrations here?